### PR TITLE
1.5 test cleanup

### PR DESCRIPTION
**Pull request by arschmitz**
This breaks the grunt config down into individual modules making it easier to read and work on
It also removes most of the previous grunt task alias and replaces with a few basic ones and one magic ci one that changes based on environment variables to allow us to only do a full build and test demos once in its own build. With the addition of caching this greatly improves test speed on travis.

Other changes include removal of old unused build components , making the exclude pattern for suites actually work now using `-` instead of `!` since that cant work from the command line even though this was documented. It also makes the exclude patterns actually work for paths and not just suite names. 

The now working exclude pattern is now used to exclude some of the sequence tests on travis since they have all of a sudden started failing for now apparent reason only on travis.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,49 +5,30 @@ php:
...
-    JQUERIES=1.11.1
-  - TYPES=integration
-    SUITES=navigation
-    JQUERIES=2.1.1
-  - TYPES=integration
-    SUITES=navigation
-    JQUERIES=git
+  - JQUERIES=1.8.3
+    CITYPE=test
+    SUITES=-navigation/sequence/sequence-dialog-hash-key-tests.html,-navigation/sequence/sequence-push-ate-disabled-path1-path2-dialog-hash-key-tests.html
```
Comment (gabrielschulhof) [inline .travis.yml]: ...-push-**st**ate-disabled-...

**Comment by gabrielschulhof** [comment]
You dropped support for --modules, right? At least, I'm not seeing build/tasks/modules.js ... or is that because 1.5-dev is not up-to-speed wrt. master?

**Comment by arschmitz** [comment]
apparently i did drop modules but not intentionally just got lost in the shuffle ill fix it.

**Comment by arschmitz** [comment]
this landed in 1.5-dev

**Pull request closed without merging**